drivers/haptics/drv2605.c

```diff
@@ -0,0 +1,621 @@
+		}
+
+		retries--;
+
+		k_usleep(10000);
+	}
+
+	return -ETIMEDOUT;
+}
+
+static int drv2605_check_devid(const struct device *dev)
+{
+	const struct drv2605_config *config = dev->config;
+	uint8_t value;
+	int ret;
+
+	ret = i2c_reg_read_byte_dt(&config->i2c, DRV2605_REG_STATUS, &value);
+	if (ret < 0) {
+		return ret;
+	}
+
+	value = FIELD_GET(DRV2605_DEVICE_ID, value);
+
+	switch (value) {
+	case DRV2605_DEVICE_ID_DRV2605:
+	case DRV2605_DEVICE_ID_DRV2605L:
+		break;
+	default:
+		LOG_ERR("Invalid device ID found");
+		return -ENOTSUP;
+	}
+
+	LOG_DBG("Found DRV2605, DEVID: 0x%x", value);
+
+	return 0;
+}
+
+static int drv2605_gpio_config(const struct device *dev)
+{
+	const struct drv2605_config *config = dev->config;
+	int ret;
+
+	if (config->en_gpio.port != NULL) {
+		if (!gpio_is_ready_dt(&config->en_gpio)) {
+			return -ENODEV;
+		}
+
+		ret = gpio_pin_configure_dt(&config->en_gpio, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			return ret;
+		}
+	}
+
+	if (config->in_trig_gpio.port != NULL) {
+		if (!gpio_is_ready_dt(&config->in_trig_gpio)) {
+			return -ENODEV;
+		}
+
+		ret = gpio_pin_configure_dt(&config->in_trig_gpio, GPIO_OUTPUT_INACTIVE);
+		if (ret < 0) {
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+static int drv2605_init(const struct device *dev)
+{
+	const struct drv2605_config *config = dev->config;
+	struct drv2605_data *data = dev->data;
+	int ret;
+
+	data->dev = dev;
+
+	k_usleep(DRV2605_POWER_UP_DELAY_US);
+
+	if (!i2c_is_ready_dt(&config->i2c)) {
+		return -ENODEV;
+	}
+
+	k_work_init(&data->rtp_work, drv2605_rtp_work_handler);
+
+	ret = drv2605_gpio_config(dev);
+	if (ret < 0) {
+		LOG_DBG("Failed to allocate GPIOs: %d", ret);
+		return ret;
+	}
+
+	ret = drv2605_check_devid(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = drv2605_reset(dev);
+	if (ret < 0) {
+		LOG_DBG("Failed to reset device: %d", ret);
+		return ret;
+	}
+
+	ret = drv2605_hw_config(dev);
+	if (ret < 0) {
+		LOG_DBG("Failed to configure device: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static const struct haptics_driver_api drv2605_driver_api = {
+	.start_output = &drv2605_start_output,
+	.stop_output = &drv2605_stop_output,
+};
+
+#define HAPTICS_DRV2605_DEFINE(inst)                                                               \
+                                                                                                   \
+	static const struct drv2605_config drv2605_config_##inst = {                               \
+		.i2c = I2C_DT_SPEC_INST_GET(inst),                                                 \
+		.en_gpio = GPIO_DT_SPEC_INST_GET_OR(inst, ti_en_gpios, {}),                        \
+		.in_trig_gpio = GPIO_DT_SPEC_INST_GET_OR(inst, ti_in_trig_gpios, {}),              \
+		.feedback_brake_factor = DT_INST_ENUM_IDX_OR(inst, ti_feedback_brake_factor, 3),   \
+		.loop_gain = DT_INST_ENUM_IDX_OR(inst, ti_loop_gain, 2),                           \
+		.actuator_mode = DT_INST_ENUM_IDX_OR(inst, ti_actuator_mode, 0),                   \
```

Comment:
None of these properties exist in the binding, either due to bad names (start with `ti, ...` here but not in the yaml file), or not even defined at all in the yaml file (ex. `en-gpios`), so there is a lot of dead code and potentially a very stuck drv2605 when the EN input pin actually needs to be driven high to even be active :)

Note that i don't think we need/want `ti,` prefix in the first place